api/v1beta1/bgppeer_webhook.go

```diff
@@ -82,10 +82,10 @@ func (bgpPeer *BGPPeer) ValidateDelete() error {
 func (bgpPeer *BGPPeer) validateBGPPeer(existingBGPPeersList *BGPPeerList, bgpFrrMode bool) error {
 	var allErrs field.ErrorList
 
-	if err := bgpPeer.validateBGPPeersRouterID(existingBGPPeersList); err != nil {
-		allErrs = append(allErrs, err)
-	}
 	if bgpFrrMode {
+		if err := bgpPeer.validateBGPPeersRouterID(existingBGPPeersList); err != nil {
```

Comment:
Can we pass the bgpFrrMode to the validating function and skip only the different routerid part in case of frr?

Reply:
validateBGPPeersRouterID() is only called if in frr mode